backend/GymBeam/CommandHandlers/UpdateUserRoleCommandHandler.cs

```diff
@@ -0,0 +1,33 @@
+﻿using Domain;
+using GymBeam.Commands;
+using GymBeam.Exceptions;
+using MediatR;
+
+namespace GymBeam.CommandHandlers
+{
+    public class UpdateUserRoleCommandHandler : IRequestHandler<UpdateUserRoleCommand, Unit>
+    {
+        private readonly IRepository _repository;
+
+        public UpdateUserRoleCommandHandler(IRepository repository)
+        {
+            _repository = repository;
+        }
+
+        public Task<Unit> Handle(UpdateUserRoleCommand request, CancellationToken cancellationToken)
+        {
+            User user;
+            try
+            {
+                user = _repository.GetById<User>(request.UserId);
+                user.Role = request.NewRole;
```

Comment:
there should be check if new role is domain role. There should be some kind of enum in Domain of possibles roles that you should try to cast from request string to domain enum